airbyte-integrations/connectors/source-airtable/oauth2.patch.json

```diff
@@ -0,0 +1,8 @@
+{
+    "provider": "airtable",
+    "authUrlTemplate": "https://airtable.com/oauth2/v1/authorize?client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&response_type=code&state={{#urlencode}}{{{ state }}}{{/urlencode}}&scope=data.records:read%20data.recordComments:read%20schema.bases:read    ",
+    "accessTokenUrlTemplate": "https://airtable.com/oauth2/v1/token",
+    "accessTokenBody": "grant_type=authorization_code&client_id={{#urlencode}}{{{ client_id }}}{{/urlencode}}&client_secret={{#urlencode}}{{{ client_secret }}}{{/urlencode}}&redirect_uri={{#urlencode}}{{{ redirect_uri }}}{{/urlencode}}&code={{#urlencode}}{{{ code }}}{{/urlencode}}",
```

Comment:
According to [Airtabledocumentation]( https://airtable.com/developers/web/api/oauth-reference#token-creation-request) this request does not take `client_secret`, instead it takes a header (we would put that in `accessTokenHeaders`) with encodes both `client_id` and `client_secret`.

This is assuming we do have a `client_secret` for AirTable. Have we created an OAuth2 app on AirTable yet? That would help us decide how to implement the oauth2 flow here.

Reply:
Apparently AirTable also has additional parameters [`code_challenge` and `code_verifier`](https://airtable.com/developers/web/api/oauth-reference#authorization-parameter-rules) which we will need to generate cryptographically and send (apparently there is an extension to OAuth2 protocol called PKCE which requires these).

This needs to be implemented in our flow, we currently don't support this PKCE extension as part of our OAuth flow. I will create an issue for this, needs work on front-end as well.

Reply:
@mdibaiee yes I created an app on Airtable so we have a client_id and client_secret that we can use.